docs/src/main/sphinx/admin/properties-http-client.rst

```diff
@@ -0,0 +1,258 @@
+``oauth2-jwk`` to the ``http-client.http2.enabled``, and setting the value to
+``true``:
+
+.. code-block::
+
+    oauth2-jwk.http-client.http2.enabled=true
+
+The following prefixes are supported:
+
+* ``oauth2-jwk`` for :doc:`/security/oauth2`
+* ``jwk`` for :doc:`/security/jwt`
+
+``http-client.connect-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-duration`
+* **Minimum value:** ``0ms``
+* **Default value:** ``5s``
+
+Set the timeout value for establishing the connection to the external service.
+
+``http-client.http-proxy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the host and port for an HTTP proxy.
+
+``http-client.http-proxy.secure``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``false``
+
+Enable HTTPS for the proxy.
+
+``http-client.https.excluded-cipher``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Override the default excluded ciphers formatted as regex.
+
+``http-client.https.hostname-verification``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``true``
+
+Verify that the server hostname matches the server DNS name in the
+SubjectAlternativeName (SAN) field of the certificate.
+
+``http-client.https.included-cipher``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Override the default included ciphers formatted as regex.
+
+``http-client.idle-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-duration`
+* **Minimum value:** ``0ms``
+* **Default value:** ``1m``
+
+Set the maximum allowed idle time for a connection.
+
+``http-client.key-store-password``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the password for the keystore.
+
+``http-client.key-store-path``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the file path on the server to the keystore file.
+
+``http-client.log.buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Minimum value:** ``1MB``
+* **Default value:** ``1GB``
+
+Set file size for data transfer log?
+
+``http-client.log.compression-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``true``
+
+Enable compression of log data?
+
+``http-client.log.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``false``
+
+Enable the logging protocol?
+
+``http-client.log.flush-interval``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-duration`
+* **Default value:** ``10s``
+
+Set frequency of log flushing to disk?
+
+``http-client.log.max-history``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Default value:** ``15``
+
+Set the maximum file size of a history log?
```

Comment:
Limits number of logging periods (i.e. days) kept. Http client creates a new log file for each logging period, files older then `max-history` will be deleted.